st2common/st2common/models/api/pack.py

```diff
@@ -117,27 +126,46 @@ class PackAPI(BaseAPI):
                 'type': 'object',
                 'description': 'Specification for the system components and packages '
                                'required for the pack.',
-                'properties': {
-                    'stackstorm': {
-                        'type': 'string',
-                        'description': 'Required StackStorm version. Examples: ">1.6", '
-                                       '">=1.8dev, <2.0"',
-                        'pattern': ST2_VERSION_REGEX,
-                    }
-                }
+                'default': {}
             }
         }
     }
 
     @classmethod
     def to_model(cls, pack):
-        parameters = pack.__dict__
-        parameters['keywords'] = parameters.get('keywords', [])
-        parameters['files'] = parameters.get('files', [])
-        parameters['dependencies'] = parameters.get('dependencies', [])
-        parameters['system'] = parameters.get('system', {})
-
-        model = cls.model(**parameters)
+        ref = pack.ref
+        name = pack.name
+        description = pack.description
+        keywords = getattr(pack, 'keywords', [])
+        version = str(pack.version)
+
+        # Note: If some version values are not explicitly surrounded by quotes they are recognized
+        # as numbers so we cast them to string
+        if getattr(pack, 'version', None):
+            version = str(pack.version)
+
+        # Special case for old version which didn't follow semver format (e.g. 0.1, 1.0, etc.)
+        # In case the version doesn't match that format, we simply append ".0" to the end (e.g.
+        # 0.1 -> 0.1.0, 1.0, -> 1.0.0, etc.)
```

Comment:
Re validation - we still perform validation after the version is normalized.

As far as the actual normalization goes - I'm usually for being explicit and require user to specify a correct version, but it sounded that @lakshmi-kannan and @m4dcoder were against it. I do see some value in this approach since it makes migration a bit easier.